presto-main/src/main/java/com/facebook/presto/memory/ClusterMemoryManager.java

```diff
@@ -338,6 +333,30 @@ public synchronized void process(Iterable<QueryExecution> runningQueries, Suppli
         updateNodes(assignmentsRequest);
     }
 
+    public synchronized void checkForLeaks(Supplier<List<BasicQueryInfo>> allQueryInfoSupplier)
+    {
+        if (!enabled) {
+            return;
+        }
+
+        Map<QueryId, Optional<BasicQueryInfo>> allRunningQueries;
+        if (memoryManagerService.isPresent()) {
+            // We are in the multi-coordinator codepath, and thus care about the globally running queries
+            allRunningQueries = getClusterInfo(GENERAL_POOL)
+                            .getRunningQueries()
+                            .orElse(Collections.emptyList()).stream().collect(
+                                Collectors.toMap(Function.identity(), t -> Optional.empty()));
+        }
+        else {
+            // We are in the single coordinator setup, and thus care about the local queries. Ie, global queries
+            // does not make sense.
+            allRunningQueries = Maps.uniqueIndex(
+                    allQueryInfoSupplier.get().stream().map(queryInfo -> Optional.of(queryInfo)).iterator(),
+                    queryInfo -> queryInfo.get().getQueryId());
```

Comment:
```suggestion
            allRunningQueries = Maps.uniqueIndex(
                    allQueryInfoSupplier.get().stream()
                        .map(queryInfo -> Optional.of(queryInfo))
                        .iterator(),
                    queryInfo -> queryInfo.get().getQueryId());
```

Reply:
Also, consider using a collector to build the map for consistency?

Reply:
Any suggestions on how to do that ? I would like to do that as well.